customized_list_view/models/custom_list.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2024 ooops404
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import json
+
+from lxml import etree
+
+from odoo import _, api, exceptions, fields, models
+
+
+class CustomListView(models.Model):
+    _name = "custom.list.view"
+    _description = "Custom List View"
+
+    name = fields.Char(required=True)
+    model_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    model_name = fields.Char(compute="_compute_model_name", store=True)
+    list_view_id = fields.Many2one("ir.ui.view", required=True)
+    line_ids = fields.One2many("custom.list.view.line", "custom_list_view_id")
+    original_arch = fields.Text(readonly=True)
+
+    @api.constrains("model_id")
+    def _check_unique_model(self):
+        for rec in self:
+            same_model_recs = self.env["custom.list.view"].search(
+                [
+                    ("model_id", "=", rec.model_id.id),
+                    ("id", "!=", rec.id),
+                ]
+            )
+            if same_model_recs:
+                raise exceptions.ValidationError(
+                    _(
+                        "Only one record per model is allowed. "
+                        "Please modify existing record instead."
+                    )
+                )
+
+    @api.depends("model_id")
+    def _compute_model_name(self):
+        for rec in self:
+            rec.model_name = rec.model_id.model
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res.original_arch = res.list_view_id.arch
```

Comment:
Please rename `res` variable, because `create` method return record or record set. Better name will be `records` here.
I think is wrong, `res` variable is record set then you need make like this:
```py
for record in records:
    record.original_arch = record.list_view_id.arch
```